io/module_fv3_io_def.F90

```diff
@@ -1,36 +1,126 @@
+!> @file
+!> @brief fv3 I/O related configration variables.
+!> @author Jun Wang @date 01/2017
+
+!> @brief fv3 I/O related configration variables.
+!>
+!> @author Jun Wang @date 01/2017
 module module_fv3_io_def
-!
-!*** fv3 io related configration variables
-!
-! revision history
-! 01/2017   Jun Wang    Initial code
-!
-!------------------------------------------------------------------------
-!
   use esmf, only     : esmf_maxstr
   implicit none
 
+  !> Number of processors used in forecast run.
   integer           :: num_pes_fcst
-  integer           :: wrttasks_per_group, write_groups
-  integer           :: n_group
-  integer           :: num_files
-  integer           :: nbdlphys
-  integer           :: iau_offset
-  logical           :: lflname_fulltime
-  logical           :: time_unlimited
-
-  character(len=esmf_maxstr),dimension(:),allocatable :: filename_base
-  character(len=esmf_maxstr),dimension(:),allocatable :: output_file
-
-  integer,dimension(:),allocatable     :: lead_wrttask, last_wrttask
-
-  character(len=esmf_maxstr),dimension(:),allocatable :: output_grid
-  integer,dimension(:),allocatable  :: imo,jmo
-  real,dimension(:),allocatable     :: cen_lon, cen_lat
-  real,dimension(:),allocatable     :: lon1, lat1, lon2, lat2, dlon, dlat
-  real,dimension(:),allocatable     :: stdlat1, stdlat2, dx, dy
-  integer,dimension(:),allocatable  :: ideflate, quantize_nsd, zstandard_level
+
+  !> Number of write tasks per group.  
+  integer           :: wrttasks_per_group
+
+  !> ???  
+  integer           :: write_groups 
+
+  !> ???  
+  integer           :: n_group 
+
+  !> ???  
+  integer           :: num_files 
+
+  !> ???  
+  integer           :: nbdlphys 
+
+  !> ???  
+  integer           :: iau_offset 
+
+  !> ???  
+  logical           :: lflname_fulltime 
+
+  !> ???  
```

Comment:
So the next step is to, for example, add ?

```
integer :: num_pes_fcst.            !> Number of processors used in forecast run
integer :: wrttasks_per_group   !> Number of write tasks per group
integer :: write_groups               !> Number of write groups
....
```

Reply:
These are module variables and need to be documented.

Local variables in subprograms do not need to be documented with doxygen.

Reply:
Can we define a style that we will? I much prefer the one-line version because it keeps the code more compact. 

Thanks for clarifying the local/module. I didn't look carefully enough at the diff.

Reply:
We can use the one-line version where the documentation will fit on the rest of the line, but frequently it will be longer than that.

We are using a style which matches the way we do documentation in the rest of the EMC software, including UFS_UTILS.

Reply:
BTW the way to add documentation is not to add it in code review, but to do a PR against my branch, so I don't have to cut and paste all the documentation. ;-)

Reply:
Each project, and each coder, can use variations of documentation style, as long as they do not stray too far from the way we doing things in other EMC projects.

So if you want to document like this:

```
integer :: i !< some long documentation about i
!! which continues on the next line.

```
Instead of:

```
!> some long documentation about i
!! which continues on the next line.
integer :: i 

```
Then I would be happy to get the good documentation and would not consider the formatting difference to be important.